qa/rpc-tests/mempool_tx_expiry.py

```diff
@@ -0,0 +1,205 @@
+        print "Expiryheight of persist_transparent_2:", rawtx_trans['expiryheight']
+        print "Expiryheight of persist_shielded_2:", rawtx_shield['expiryheight']
+        print "Mine 4 competing blocks on Node 2, greater than expiry height"
+        blocks = self.nodes[2].generate(4)
+        for block in blocks:
+            blk = self.nodes[2].getblock(block)
+            print "Height: {0}, Mined block txs: {1}".format(blk["height"], blk["tx"])
+        print "Connect nodes to force a reorg"
+        connect_nodes_bi(self.nodes, 0, 2)
+        self.is_network_split = False
+        sync_blocks(self.nodes)
+        print "Ensure that persist_transparent_2 & persist_shielded_2 are in mempool at expiry block height: ", persist_transparent_2, persist_shielded_2
+        print "Blockheight node 0:", self.nodes[0].getinfo()['blocks']
+        print "Blockheight node 2:", self.nodes[2].getinfo()['blocks']
+        print "mempool node 0: ", self.nodes[0].getrawmempool()
+        print "mempool node 2: ", self.nodes[2].getrawmempool()
+        print "Testing persist_transparent_2 in mempool"
+        assert(persist_transparent_2 in self.nodes[0].getrawmempool())
+        # Shielded Tx is not getting put back in mempool >:(
+        print "Testing persist_shielded_2 in mempool"
+        assert(persist_shielded_2 in self.nodes[0].getrawmempool())
+        # Mine persist txs to get them out of the way of mempool sync in split_network()
+        self.nodes[0].generate(1)
+        assert_equal(set(self.nodes[0].getrawmempool()), set())
+        sync_blocks(self.nodes)
+        print "Balance after persist_shielded_2 is mined to remove from mempool: ", self.nodes[0].z_gettotalbalance()
+
+        print "Splitting network..."
+        self.split_network()
+
+        ## AFTER expiry block height: Reorg and test case where txs expire from mempool
+        print "Balance before expire_shielded is sent: ", self.nodes[0].z_gettotalbalance()
+        myopid = self.nodes[0].z_sendmany(z_alice, recipients)
+        print "status of expire_shielded", self.nodes[0].z_getoperationstatus([myopid])
+        expire_shielded = wait_and_assert_operationid_status(self.nodes[0], myopid)
+        expire_transparent = self.nodes[0].sendtoaddress(bob, 0.01)
+        print "Blockheight node 0 at expire_transparent creation:", self.nodes[0].getinfo()['blocks']
+        print "Blockheight node 2 at expire_shielded creation:", self.nodes[2].getinfo()['blocks']
+        print "Expiryheight of expire_transparent:", self.nodes[0].getrawtransaction(expire_transparent, 1)['expiryheight']
+        print "Expiryheight of expire_shielded:", self.nodes[0].getrawtransaction(expire_shielded, 1)['expiryheight']
+        assert(expire_transparent in self.nodes[0].getrawmempool())
+        assert(expire_shielded in self.nodes[0].getrawmempool())
+        print "Mine 4 competing blocks on Node 2, greater than expiry height"
```

Comment:
Out-of-date debugging comment.

I'm finding that these debugging print statements are making the RPC test as a whole harder to review. Some new lines are definitely needed to split this up into sections, and some of the more debugging-ish prints could be removed.